src/policy/packages.h

```diff
@@ -62,4 +65,39 @@ bool IsPackageWellFormed(const Package& txns, PackageValidationState& state);
  */
 bool IsChildWithParents(const Package& package);
 
+class Packageifier
+{
+    /** Transactions sorted topologically. */
+    Package txns;
+    /** Caches the transactions by txid for quick lookup. */
+    std::map<uint256, CTransactionRef> txid_to_tx;
+    /** Caches the in-package ancestors for each transaction. */
+    std::map<uint256, std::set<uint256>> ancestor_subsets;
+    /** Transactions to exclude when returning ancestor subsets.*/
+    std::unordered_set<uint256, SaltedTxidHasher> excluded_txns;
+    /** Transactions that are banned. Return empty vector if any ancestor subset contains these transactions.*/
```

Comment:
8764ecc566413b4974e030c596193e75ff39f746
The comment is incorrect: nullopt is returned, not empty vector.

Reply:
Thanks, fixed